lib/client/doc.js

```diff
@@ -748,6 +750,15 @@ Doc.prototype._submit = function(op, source, callback) {
   // The op contains either op, create, delete, or none of the above (a no-op).
   if ('op' in op) {
     if (!this.type) {
+      if (this._isInHardRollback) {
+        var err = new ShareDBError(
+          ERROR_CODE.ERR_DOC_IN_HARD_ROLLBACK,
+          'Cannot submit op. Document is performing hard rollback. ' + this.collection + '.' + this.id
+        );
```

Comment:
You could just move this after the `var err = new ShareDBError(ERR_DOC_DOES_NOT_EXIST)` to override this "default" error and then avoid having to duplicate the `emit()` machinery